.github/workflows/testing-and-deployment.yml

```diff
@@ -205,9 +206,9 @@ jobs:
       - name: Install Testing Requirements
         run: pip install -r requirements_test.txt
 
-      - name: Limit NumPy for VTK 9.0.3
-        if: ${{ matrix.vtk-version == '9.0.3' }}
-        run: pip install 'numpy<1.24'
+      - name: Set up numpy
+        if: ${{ matrix.numpy-version != 'latest' && matrix.numpy-version != 'nightly' }}
+        run: pip install 'numpy<${{ matrix.numpy-version }}'
```

Comment:
Having `numpy < matrix.numpy-version` is a but surprising to me, even though I see that the result preserves `numpy<1.24`. So how about setting the highest known `1.23` version instead, and using `<=` or a similar operator?

Reply:
Now using ~= checks.

Reply:
I learned the hard way that `numpy~=1.23` will install version 1.26.4.  It needs to be `numpy~=1.23.0` which will give the latest patch release from 1.23.x.  This is essentially the same as `numpy<1.24`.  But the ~= usage is nicer for the CI, so the name of the job matrix is accurate.